src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -419,7 +419,7 @@ private static function normalizePeerFingerprint(mixed $fingerprint): array
      */
     private static function jsonEncode(mixed $value, int $flags = null, int $maxDepth = 512): string
     {
-        $flags ??= \JSON_HEX_TAG | \JSON_HEX_APOS | \JSON_HEX_AMP | \JSON_HEX_QUOT | \JSON_PRESERVE_ZERO_FRACTION;
+        $flags |= \JSON_HEX_TAG | \JSON_HEX_APOS | \JSON_HEX_AMP | \JSON_HEX_QUOT | \JSON_PRESERVE_ZERO_FRACTION;
```

Comment:
Do we have existing code that passes a non-null value for `$flags` ? If yes, what is the impact on that code if we add those flags to the provided one ?

Reply:
The only other call on `jsonEncode` is inside the HttpClientTrait itself, without flags. So we can consider that we break nothing here.